packages/data-broker/src/services/dhis/DhisInputSchemeResolvingApi.js

```diff
@@ -0,0 +1,306 @@
+    //   if (dataElement.config.dhisId) {
+    //     return true;
+    //   }
+    // }
+
+    for (const dataElementCode of dataElementCodes) {
+      if (DATA_SOURCE_CONFIG_BY_CODE[dataElementCode]) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  /**
+   * @param query {*}
+   * @returns bool
+   * @private
+   */
+  anyOrgUnitsHaveDhisId = async query => {
+    const { organisationUnitCode, organisationUnitCodes } = query;
+
+    const orgUnits = await this.models.entity.find({
+      code: organisationUnitCode || organisationUnitCodes,
+    });
+
+    for (const orgUnit of orgUnits) {
+      if (TABLE_DHIS_ORG_UNIT_MAPPING[orgUnit.code]) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  /**
+   * @param query {*}
+   * @returns bool
+   * @private
+   */
+  anyProgramsHaveDhisId = async query => {
+    const { programCode } = query;
+
+    // const dataElements = await this.models.dataSource.find({ code: programCode });
+    //
+    // for (const dataElement of dataElements) {
+    //   if (dataElement.config.dhisId) {
+    //     return true;
+    //   }
+    // }
+
+    if (DATA_SOURCE_CONFIG_BY_CODE[programCode]) {
+      return true;
+    }
+
+    return false;
+  };
+
+  /**
+   * @param query {*}
+   * @returns {*}
+   * @private
+   */
+  replaceDataElementCodesWithIds = async query => {
+    const modifiedQuery = { ...query, dataElementIds: [] };
+
+    const { dataElementCodes } = query;
+
+    // const dataElements = await this.models.dataSource.find({ code: dataElementCodes });
+
+    for (const dataElementCode of dataElementCodes) {
+      // const dataElement = dataElements.find(d => d.code === dataElementCode);
+      //
+      // if (!dataElement) {
+      //   throw new Error(
+      //     'DataElement not found in data_source, attempted to replace its code with an id',
+      //   );
+      // }
+      //
+      // if (!dataElement.config.dhisId) {
+      //   throw new Error(
+      //     'DataElement does not have a dhisId, attempted to replace its code with the id',
+      //   );
+      // }
+      //
+      // modifiedQuery.dataElementIds.push(dataElement.config.dhisId);
+
+      if (!DATA_SOURCE_CONFIG_BY_CODE[dataElementCode]) {
+        throw new Error(
+          'DataElement not found in data_source, attempted to replace its code with an id',
+        );
+      }
+
+      if (!DATA_SOURCE_CONFIG_BY_CODE[dataElementCode].dhisId) {
+        throw new Error(
+          'DataElement does not have a dhisId, attempted to replace its code with the id',
+        );
+      }
+
+      modifiedQuery.dataElementIds.push(DATA_SOURCE_CONFIG_BY_CODE[dataElementCode].dhisId);
+    }
+
+    delete modifiedQuery.dataElementCodes;
+
+    return modifiedQuery;
+  };
+
+  /**
+   * @param query {*}
+   * @returns {*}
+   * @private
+   */
+  replaceOrgUnitCodesWithIds = async query => {
+    const modifiedQuery = { ...query, organisationUnitIds: [] };
```

Comment:
Note to self: if we set dhisId on these entities and dataElements, and this proxy runs into an api request that hits any of them, it will require them all to have ids. 

TODO: change the logic here to: if they don't _all_ have ids, use codes instead.

In the case of laos eoc, when building dashboards and map overlays, we will know pretty quickly if we have a missing dhisId, because the whole thing will fall over.